src/install.rs

```diff
@@ -320,35 +321,15 @@ async fn remove_package_from_environment(
     target_prefix: &Path,
     package: &PrefixRecord,
 ) -> miette::Result<()> {
-    // TODO: Take into account any clobbered files, they need to be restored.
-    // TODO: Can we also delete empty directories?
-
-    // Remove all entries
-    for paths in package.paths_data.paths.iter() {
-        match tokio::fs::remove_file(target_prefix.join(&paths.relative_path)).await {
-            Ok(_) => {}
-            Err(e) if e.kind() == ErrorKind::NotFound => {
-                // Simply ignore if the file is already gone.
-            }
-            Err(e) => {
-                return Err(e).into_diagnostic().wrap_err(format!(
-                    "failed to delete {}",
-                    paths.relative_path.display()
-                ))
-            }
+    match unlink_package(target_prefix, package).await {
+        Ok(_) => Ok(()),
+        Err(err) => {
+            println!("\n\n\nFailed to remove package: {}", err);
+            miette::bail!(
+                "Failed to remove package {}: {}",
+                package.repodata_record.file_name,
+                err
+            );
```

Comment:
I guess bailing might even be a lot. Wouldn't it be better to return an error which could then be reported as a warning perhaps. Or is this a really big issue?

If so what should the user do when this happens?

Reply:
Apparently, `bail!` returns early with an error (which is what we want in this case).

https://docs.rs/miette/latest/miette/macro.bail.html